v22.2/known-limitations.md

```diff
@@ -10,6 +10,85 @@ This page describes newly identified limitations in the CockroachDB {{page.relea
 
 ## New limitations
 
+### DROP OWNED BY does not support drop functions
+
+`DROP OWNED BY` drops all owned objects as well as any grants on objects not owned by the role.
+
+In its current implementation, this statement does not drop functions. Users must drop their functions manually.
+
+[Tracking GitHub Issue](https://github.com/cockroachdb/cockroach/issues/90476)
+
+{% comment %}
+    initial DROP OWNED BY implementation PR/issue:
+    https://github.com/cockroachdb/cockroach/pull/82936
+    https://github.com/cockroachdb/cockroach/issues/55381
+    doc work in progress (to link to after merge):
+    https://github.com/cockroachdb/docs/pull/15094/files
+    https://cockroachlabs.atlassian.net/browse/DOC-4770 
+    review: rafiss (or devadvocado)
+{% endcomment %}
+
+### DROP OWNED BY not supported where role has synthetic privileges
+
+`DROP OWNED BY` drops all owned objects as well as any grants on objects not owned by the role.
+
+In its current implentation, this operation cannot be performed for roles that have synthetic privileges (entries in `system.privileges`).
+
+Instead, you can use `REVOKE SYSTEM` for the relevant privileges the role has in `system.privileges`.
+
+[Tracking GitHub Issue](https://github.com/cockroachdb/cockroach/issues/88149)
+
+{% comment %}
+    initial DROP OWNED BY implementation PR/issue:
+    https://github.com/cockroachdb/cockroach/pull/82936
+    https://github.com/cockroachdb/cockroach/issues/55381
+    doc work in progress (to link to after merge):
+    https://github.com/cockroachdb/docs/pull/15094/files
+    https://cockroachlabs.atlassian.net/browse/DOC-4770 
+    review: rafiss
+{% endcomment %}
+
+### Expressions with `*` not supported in user defined functions (UDFs)
+
+Expressions such as `SELECT *` are not currently allowed within the body of a UDF.
+
+[Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/90080)
+
+{% comment %}
+    review: mgartner
+    see also https://github.com/cockroachdb/cockroach/issues/86070
+    Link to UDF docs?
+{% endcomment %}
+
+### PARTITION ALL BY can lead to poor performance
+
+`PARTITION ALL BY` was added to support multi-region abstractions. In that setting it typically performs well due to locality optimized search. In other cases, unique indexes can result in poor performance because uniqueness validation may need to perform full index scans. Additionally, creating an index does not mean that there will be any acceleration on queries constraining that column.
+
+[Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/83419)
```

Comment:
@ajwerner Could you please review these lines—this one heading and paragraph?

It's an edit of something you wrote in one of the relevant issues. I think the mention of "In other cases" is a bit unclear - e.g. are we referring to 'cases other than those pertaining to multi-region abstractions?

Any edits for clarity (and correctness) would be appreciated.

Reply:
The message we should convey, to the extent we convey anything, is that `PARTITION ALL BY` is not a feature users should generally use unless they really know what they are doing. You have to opt into it to use it. I'm not so sure we have to have it listed as a known-limitation given it's gated behind `experimental_enable_implicit_column_partitioning`.

Reply:
With that context, I lean against including this, especially given that I'm not seeing this feature documented. It is merely mentioned under [Hash-sharded indexes on partitioned tables](https://www.cockroachlabs.com/docs/stable/hash-sharded-indexes.html#hash-sharded-indexes-on-partitioned-tables) and otherwise in some release notes. I will ask for the SQL Schema PM's take on this as well.

Worth noting that at the above link, the feature is referred to as being "in preview", but in preview, per recent policy, a feature should have docs.

FYI for @lnhsingh: I'm not sure if this is a case where a feature is truly more "experimental" and we're constrained by the new policy of exclusively using the term "preview", or if it can correctly be called preview and should have docs. If the former, perhaps we need to update our policy - either carving out criteria where we can bring back the experimental classification OR criteria where we won't document preview features. Will flag this for us to discuss.

Reply:
Update: Decided to leave this out after consulting Lauren. Thanks.